webapp/sources/rudder/rudder-web/src/main/scala/com/normation/rudder/web/snippet/node/NodeHistoryViewer.scala

```diff
@@ -67,19 +70,27 @@ class NodeHistoryViewer extends StatefulSnippet {
   var dispatch: DispatchIt = { case "render" => render _ }
 
   def render(xml: NodeSeq): NodeSeq = {
+
+    implicit val qc: QueryContext = CurrentUser.queryContext
+
     S.attr("uuid") match {
       case Full(s) => // new id of id change, init for that id
         initState(s)
 
         <div>
           <p>{SHtml.ajaxSelectObj[DateTime](dates, Full(selectedDate), onSelect _)}</p>
           {
-          historyRepos.get(uuid, selectedDate).toBox match {
-            case Failure(m, _, _)   => <div class="error">Error while trying to display node history. Error message: {m}</div>
-            case Empty | Full(None) => <div class="error">No history was retrieved for the chosen date</div>
-            case Full(Some(sm))     =>
+          (for {
+            globalMode <- configService
+                            .rudder_global_policy_mode()
+                            .chainError(s" Could not get global policy mode when getting node '${uuid.value}' details")
+            data       <- historyRepos.get(uuid, selectedDate)
+          } yield (globalMode, data)).toBox match {
```

Comment:
impact of previous change about really wanting the global policy mode: it needs to be retrieved here, too.